pkg/controller/nodeipam/ipam/cidr_allocator.go

```diff
@@ -144,3 +169,34 @@ func listNodes(kubeClient clientset.Interface) (*v1.NodeList, error) {
 	}
 	return nodeList, nil
 }
+
+func listClusterCIDRConfigs(kubeClient clientset.Interface) (*networkingv1alpha1.ClusterCIDRConfigList, error) {
+	var clusterCIDRConfigList *networkingv1alpha1.ClusterCIDRConfigList
+	// We must poll because apiserver might not be up. This error causes
+	// controller manager to restart.
+	if pollErr := wait.Poll(clusterCIDRConfigPollInterval, apiserverStartupGracePeriod, func() (bool, error) {
```

Comment:
Let's add info about how long it was waiting to make it easier for people debugging to see how long it waited.

startTimestamp := time.Now()

klog.Errorf("Failed to list clusterCIDRConfigs (after %v)", time.Now().Sub(startTimestamp))

We might also want to use a exp backoff with a small maximum instead of a linear poll

Reply:
Done